UnitTests/GitCommandsTests/Git/ObjectIdTests.cs

```diff
@@ -0,0 +1,155 @@
+        [TestCase("000000000000000000000000000000000000000")]
+        [TestCase("01020304050607080910111213141516171819200")]
+        [TestCase("010203040506070809101112131415161718192001")]
+        [TestCase("ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ")]
+        [TestCase("  0000000000000000000000000000000000000000  ")]
+        public void TryParse_identifies_invalid_hashes(string sha1)
+        {
+            Assert.False(ObjectId.TryParse(sha1, out _));
+        }
+
+        [TestCase("0000000000000000000000000000000000000000")]
+        [TestCase("0102030405060708091011121314151617181920")]
+        [TestCase("abcdefABCEF0123456789abcdefABCDEF0123456")]
+        public void Parse_handles_valid_hashes(string sha1)
+        {
+            Assert.AreEqual(
+                sha1.ToLower(),
+                ObjectId.Parse(sha1).ToString());
+        }
+
+        [TestCase("00000000000000000000000000000000000000")]
+        [TestCase("000000000000000000000000000000000000000")]
+        [TestCase("01020304050607080910111213141516171819200")]
+        [TestCase("010203040506070809101112131415161718192001")]
+        [TestCase("ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ")]
+        [TestCase("  0000000000000000000000000000000000000000  ")]
+        public void Parse_throws_for_invalid_hashes(string sha1)
+        {
+            Assert.Throws<FormatException>(() => ObjectId.Parse(sha1));
+        }
+
+        [TestCase("0000000000000000000000000000000000000000")]
+        [TestCase("0102030405060708091011121314151617181920")]
+        [TestCase("abcdefABCEF0123456789abcdefABCDEF0123456")]
+        public void IsValid_identifies_valid_hashes(string sha1)
+        {
+            Assert.True(ObjectId.IsValid(sha1));
+        }
+
+        [TestCase("00000000000000000000000000000000000000")]
+        [TestCase("000000000000000000000000000000000000000")]
+        [TestCase("01020304050607080910111213141516171819200")]
+        [TestCase("010203040506070809101112131415161718192001")]
+        [TestCase("ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ")]
+        [TestCase("  0000000000000000000000000000000000000000  ")]
+        public void IsValid_identifies_invalid_hashes(string sha1)
+        {
+            Assert.False(ObjectId.IsValid(sha1));
+        }
+
+        [Test]
+        public void UnstagedId_has_expected_value()
+        {
+            Assert.AreEqual(
+                "0000000000000000000000000000000000000000",
+                ObjectId.UnstagedId.ToString());
+        }
+
+        [Test]
+        public void IndexId_has_expected_value()
+        {
+            Assert.AreEqual(
+                "1111111111111111111111111111111111111111",
+                ObjectId.IndexId.ToString());
+        }
+
+        [Test] public void UnstagedId_is_artificial()      => Assert.IsTrue(ObjectId.UnstagedId.IsArtificial);
+        [Test] public void IndexId_is_artificial()         => Assert.IsTrue(ObjectId.IndexId.IsArtificial);
```

Comment:
Please turn these into full-body methods to be consistent with the rest.

Reply:
Sure.